File: Week4/18-7_최윤성_20210730.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int main() {
    priority_queue<pair<int, int>, vector<pair<int, int> >, greater<pair<int, int> > > time;
    int maxCount = 0;
    int n;
    cin >> n;
    for(int i = 0; i < n; i++) {
        int start, end;
        cin >> start >> end;
        time.push(make_pair(end, start));
    }

    for(int next = 0; !time.empty(); time.pop()) {
        if(time.top().second >= next) {
            next = time.top().first;
            maxCount++;
        }
    }
    cout << maxCount << endl;
    return 0;
}

File: Week4/18-7_최윤성_20210730.py
n = int(input())
time = []
maxCount = 0
next = 0
for _ in range(n):
    time.append(list(map(int, input().split()[::-1])))
for end, start in sorted(time):
    if start >= next:
        next = end
        maxCount += 1
print(maxCount)